Mathlib/Analysis/SpecialFunctions/Complex/Log.lean

```diff
@@ -255,51 +255,49 @@ variable {α ι: Type*}
 
 open Real
 
-lemma Real.HasSum_rexp_HasProd (f : ι → α → ℝ) (hfn : ∀ x n, 0 < f n x)
-    (hf : ∀ x : α, HasSum (fun n => log (f n x)) (∑' i, log (f i x))) (a : α) :
-       HasProd (fun b ↦ f b a) (∏' n : ι, (f n a)) := by
-  have : HasProd (fun b ↦ f b a) ((rexp ∘ fun a ↦ ∑' (n : ι), log (f n a)) a) := by
-    apply ((hf a).rexp).congr
+lemma Real.HasSum_rexp_HasProd (f : ι → ℝ) (hfn : ∀ n, 0 < f n)
+    (hf : HasSum (fun n => log (f n)) (∑' i, log (f i))) :
```

Comment:
The shape of this hypothesis looks a little weird to me, isn't it just saying that this is summable? That is, it currently says "it has sum given by its tsum", which should be the same thing as just saying "it is summable". And similarly for the output type

Reply:
So I think this is common in the summable/multipliable stuff? In any case, when I originally added these results I was asked to add the `HasProd` versions #15378

Reply:
Having a `HasProd` version is good, but typically that version isn't written using `tprod` like it is here. I'll be more specific in just a moment

Reply:
Hmm I think I see what you maybe mean. Is it better now?

Reply:
Ah, you beat me to it! I will double-check but my first glance looked like what I was going to do, yeah